addon/modules/object-browser-serializer.js

```diff
@@ -0,0 +1,198 @@
+      return
+    }
+    let activeSorting = sortQueryParam.map(sortItem => {
+      if (sortItem.indexOf('-') === -1) {
+        return {
+          value: sortItem,
+          direction: ':asc'
+        }
+      } else {
+        return {
+          value: sortItem.slice(1),
+          direction: ':desc'
+        }
+      }
+    })
+    controller.set('objectBrowserConfig.listConfig.sorting.active', activeSorting)
+  },
+
+  query (params) {
+    const context = this.get('context')
+    console.log(params.filterQueryParam)
+    console.log(params.sortQueryParam)
+    console.log(params.pageQueryParam)
+
+    let queryObject = {
+      filter: params && params.filterQueryParam,
+      sort: params && params.sortQueryParam,
+      page: params && params.pageQueryParam
+    }
+
+    const serializedQueryObject = this.serializeQueryParams.call(context, queryObject)
+    this.willQuery.call(context, serializedQueryObject)
+    if (true) {
+      return this.serverQuery(serializedQueryObject, context).then(
+        response => {
+          // get pagination module
+          this.get('context').clearListState()
+          let paginationHelper = this.get('pagination')
+          return paginationHelper ? paginationHelper.processPageResponse(response, context, queryObject) : response
+        },
+        error => {
+          this.queryErrorHandler(error)
+        })
+    } else {
+      let dataKey = context.get('objectBrowserConfig.listConfig.items')
+      let result = context.get(dataKey)
+      if (this.get('config.filter.client')) {
+        result = this.clientFilter(result, queryObject.filter)
+      }
+      if (this.get('config.sort.client')) {
+        result = this.clientSort(result, queryObject.sort)
+      }
+      this.didQuery.call(context, serializedQueryObject)
+      return Ember.RSVP.resolve(result)
+    }
+  },
+
+  serverQuery (queryObject, context) {
+    let promise = context.get('store').query(this.get('config.model'), queryObject).then(
+      (response) => {
+        let processedResponse = this.didReceiveResponse.call(context, response)
+
+        let meta = response.get('meta')
+
+        if (this.get('config.filter.client')) {
+          processedResponse = this.clientFilter(processedResponse, queryObject.filter)
+        }
+        if (this.get('config.sort.client') && queryObject.sort.length > 0) {
+          processedResponse = this.clientSort(processedResponse, queryObject.sort)
+        }
+
+        Ember.set(processedResponse, 'meta', meta)
+        return processedResponse
+      }
+    )
+    this.didQuery.call(context, queryObject)
+    return promise
+  },
+
+  /*
+   Function runs when
+   filter: {
+     client: true / client: <function>
+   },
+   */
+  clientFilter (items, filter) {
+    const context = this.get('context')
+    const config = context.get('objectBrowserConfig.serializerConfig.filter.client')
+    if (config) {
+      if (config && typeof config === 'function') {
```

Comment:
why are we checking config twice here once in the first if and another time in the second if. From my understanding we should probably have something like:
```
if (config) {
  if (typeof config === 'function') {
    ...
  } else {
    ...
  }
}
```